debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/snapshot/QueryingSnapshotter.java

```diff
@@ -44,8 +44,16 @@ public String snapshotTransactionIsolationLevelStatement(SlotCreationResult newS
              * For an on demand blocking snapshot we don't need to reuse
              * the same snapshot from the existing exported transaction as for the initial snapshot.
              */
-            String snapSet = String.format("SET TRANSACTION SNAPSHOT '%s';", newSlotInfo.snapshotName());
-            return "SET TRANSACTION ISOLATION LEVEL REPEATABLE READ; \n" + snapSet;
+            try {
+                Thread.sleep(1000);
```

Comment:
Why do we sleep here? Please add comment with an explanation.

Reply:
This is a temporary change.

The consistent snapshot time is set as the upper bound of the maximum time on the nodes of the Universe and could be 0.5 seconds ahead of the time on some of the t-server nodes. The "SET LOCAL yb_read_time" will return an error if the time to be set is in the future. The sleep for 1 second to ensure that this does not happen.

Most likely this will be fixed on the YB server side. At that point, this sleep can be removed from here.